### PR TITLE
build: cbindgen (v11)

### PR DESCRIPTION
**Pull request by dbcfd**
Version 11 (from https://github.com/OISF/suricata/pull/4291)

Rust headers are now generated using cbindgen. If cbindgen is present, they can
be generated during dist, otherwise they will be available for builds.

Closes https://redmine.openinfosecfoundation.org/issues/3308

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Rust headers are now created by cbindgen
- Cbindgen is run as a step under dist when cbindgen is available and a high enough version

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -2383,25 +2383,17 @@ fi
 
     rust_vendor_comment="# "
     have_rust_vendor="no"
+    rust_compiler_version="not set"
+    rust_cargo_version="not set"
```
Comment (jasonish) [inline configure.ac]: These are set already at line 2365.

File: rust/Makefile.am [hunk 2]
```diff
@@ -42,12 +42,30 @@ else
 			--features "$(RUST_FEATURES)"
 endif
 
+if HAVE_RUST_HEADERS
+
+all-local: build-rust
+	$(info Rust headers available)
```
Comment (jasonish) [inline rust/Makefile.am]: Will this only work on GNU make?

Comment (jasonish) [reply rust/Makefile.am]: > Will this only work on GNU make?

While it doesn't fail on FreeBSD, it doesn't print anything either. Probably best to stick with @echo...

**Comment by dbcfd** [comment]
Closing for https://github.com/OISF/suricata/pull/4385

**Pull request closed without merging**